lib/fizzy/parser_expr.cpp

```diff
@@ -170,6 +170,54 @@ parser_result<Code> parse_expr(const uint8_t* pos)
             break;
         }
 
+        case Instr::if_:
+        {
+            const uint8_t type{*pos};
+            uint8_t arity;
+            if (type == BlockTypeEmpty)
+            {
+                arity = 0;
+                ++pos;
+            }
+            else
+            {
+                // Will throw in case of incorrect type.
+                std::tie(std::ignore, pos) = parser<ValType>{}(pos);
+                arity = 1;
+            }
+
+            code.immediates.push_back(arity);
+
+            label_positions.push_back(code.immediates.size());
+
+            // Placeholders for immediate values, filled at the matching end and else instructions.
+            push(code.immediates, uint32_t{0});  // Diff to the end instruction.
+            push(code.immediates, uint32_t{0});  // Diff for the immediates
+
+            push(code.immediates, uint32_t{0});  // Diff to the else instruction
+            push(code.immediates, uint32_t{0});  // Diff for the immediates.
+
+            break;
+        }
+
+        case Instr::else_:
+        {
+            if (label_positions.empty())
```

Comment:
Wouldn't this still allow `block else`?

Reply:
It would. I can add verification that we it's `if` label like it's done for `loop` and `LoopLabel` or just not to check this because it's verifier's job.

Reply:
To resolve "verification" here, the best would be to add "type" filed to `Label`. This will also help a bit with `loop` case where we use sentinel `Label` constant.

But I don't think this is good direction to more checks to parser, which is not a validator. Parser should not crash on invalid input (and this check prevents this).

I would first construct the `(block (else ...))` example to see what happens during execution.

Reply:
Any decision on that? Should I try to create bytecode with `(block (else ...))` by hand?

Reply:
I will handle this change quickly myself.

Reply:
See https://github.com/wasmx/fizzy/pull/120.

Reply:
I agree to do most of the instruction validation as per #63. However I think if it is not too hard, we should do this simple validation here. Based on #120 it seems to be trivial now.

Reply:
Implemented.